app/src/openxr/cpp/OpenXRInputSource.cpp

```diff
@@ -701,7 +701,10 @@ void OpenXRInputSource::Update(const XrFrameState& frameState, XrSpace localSpac
 
     delegate.SetLeftHanded(mIndex, mHandeness == OpenXRHandFlags::Left);
     delegate.SetTargetRayMode(mIndex, device::TargetRayMode::TrackedPointer);
-    delegate.SetControllerType(mIndex, mActiveMapping->controllerType);
+    if (mActiveMapping->controllerType == device::UnknownType)
+        delegate.SetControllerType(mIndex, DeviceUtils::GetDeviceTypeFromSystem(false));
+    else
+        delegate.SetControllerType(mIndex, mActiveMapping->controllerType);
```

Comment:
I prefer this to happen in `OpenXRInputSource::Initialize()`. We can directly modify the mapping that is added to the `mMappings` vector.

Reply:
Sure, done.